mlflow/store/model_registry/abstract_store.py

```diff
@@ -807,3 +819,66 @@ def delete_prompt_alias(self, name: str, alias: str) -> None:
             alias: Alias to delete.
         """
         self.delete_registered_model_alias(name, alias)
+
+    def link_prompt_version_to_model(self, name: str, version: str, model_id: str) -> None:
+        """
+        Link a prompt version to a model.
+
+        Default implementation sets a tag. Stores can override with custom behavior.
+
+        Args:
+            name: Name of the prompt.
+            version: Version of the prompt to link.
+            model_id: ID of the model to link to.
+        """
+        from mlflow.tracking import _get_store as _get_tracking_store
+
+        prompt_version = self.get_prompt_version(name, version)
+        tracking_store = _get_tracking_store()
+
+        with self._prompt_link_lock:
+            logged_model = tracking_store.get_logged_model(model_id)
+            if not logged_model:
+                raise MlflowException(
+                    f"Could not find model with ID '{model_id}' to which to link prompt '{name}'.",
+                    error_code=ErrorCode.Name(RESOURCE_DOES_NOT_EXIST),
+                )
+
+            prompts_tag_value = logged_model.tags.get(LINKED_PROMPTS_TAG_KEY)
```

Comment:
Setting prompts as tags on the logged model is far more scalable than setting logged models as tags on prompts. Users might use a prompt with thousands, hundreds of thousands, etc agent versions. This direction supports that use case; setting models as tags on prompts does not.